site/docs/spec.md

```diff
@@ -979,30 +979,88 @@ This serialization scheme is for storing single values as individual binary valu
 
 ## Format version changes
 
+Some tables in this spec have columns that specify requirements for v1 and v2 tables. These requirements are intended for writers when adding metadata files to a table with the given version.
+
+| Requirement | Write behavior |
+|-------------|----------------|
+| (blank)     | The field should be omitted |
+| _optional_  | The field can be written |
+| _required_  | The field must be written |
+
+Readers should be more permissive because v1 metadata files are allowed in v2 tables so that tables can be upgraded to v2 without rewriting the metadata tree. For manifest list and manifest files, this table shows the expected v2 read behavior:
+
+| v1         | v2         | v2 read behavior |
+|------------|------------|------------------|
+|            | _optional_ | Read the field as _optional_ |
+|            | _required_ | Read the field as _optional_; it may be missing in v1 files |
```

Comment:
If it is required in v2 but not present, should it be defaulted while reading?

Reply:
In most cases, yes. But reading the field requires you to set it as optional or else the reader will fail when it isn't present in the older file.

Reply:
Ok, so we essentially refer to the schema for reading the data.

Reply:
This makes sense.